src/main/kotlin/domain/Minesweeper.kt

```diff
@@ -26,5 +26,45 @@ class Minesweeper(minesweeperProperty: MinesweeperProperty, mineAllocationStrate
                 val col = it % minesweeperProperty.width
                 _board[row][col] = Place(it, PlaceType.MINE)
             }
+
+        calculateNearMineCount(minesweeperProperty)
+    }
+
+    private fun calculateNearMineCount(minesweeperProperty: MinesweeperProperty) {
+        for (row: Int in 0 until minesweeperProperty.height) {
+            for (col: Int in 0 until minesweeperProperty.width) {
+                val place = _board[row][col]
+                if (place.isNotMine()) {
+                    _board[row][col] = Place(place.number, place.placeType, getNearMineCount(row, col))
+                }
+            }
+        }
+    }
+
+    private fun getNearMineCount(row: Int, col: Int): Int {
+        var count = 0
+
+        if (isMineInPlace(row - 1, col - 1)) count++
+        if (isMineInPlace(row - 1, col)) count++
+        if (isMineInPlace(row - 1, col + 1)) count++
+        if (isMineInPlace(row, col - 1)) count++
+        if (isMineInPlace(row, col + 1)) count++
+        if (isMineInPlace(row + 1, col - 1)) count++
+        if (isMineInPlace(row + 1, col)) count++
+        if (isMineInPlace(row + 1, col + 1)) count++
+
+        return count
+    }
+
+    private fun isMineInPlace(row: Int, col: Int): Boolean {
+        return getPlaceOrNull(row, col)?.isMine() ?: false
+    }
+
+    private fun getPlaceOrNull(row: Int, col: Int): Place? {
+        return try {
+            _board[row][col]
+        } catch (e: IndexOutOfBoundsException) {
+            null
+        }
```

Comment:
유효하지 않은 위치에 대한 접근에 null을 반환하는 경우라면 아래처럼 나타내면 어떨까요?

```suggestion
        return _board.getOrNull(row)?.getOrNull(col)
```